extensions/amp-next-page/1.0/visibility-observer.js

```diff
@@ -66,86 +97,68 @@ export class VisibilityObserverEntry {
     this.observer_
       .getPositionObserver()
       .observe(top, PositionObserverFidelity.LOW, position =>
-        this.topSentinelPositionChanged(position, callback)
+        this.topSentinelPositionChanged(position)
       );
     this.observer_
       .getPositionObserver()
       .observe(bottom, PositionObserverFidelity.LOW, position =>
-        this.bottomSentinelPositionChanged(position, callback)
+        this.bottomSentinelPositionChanged(position)
       );
   }
 
   /**
    * Called when a position change is detected on the injected
    * top sentinel element
    * @param {?PositionInViewportEntryDef} position
-   * @param {function(!ViewportRelativePos)} callback
    */
-  topSentinelPositionChanged(position, callback) {
-    const prevTopSentinelPosition = this.topSentinelPosition_;
-    if (!position || position.relativePos === prevTopSentinelPosition) {
-      return;
-    }
+  topSentinelPositionChanged(position) {
     this.topSentinelPosition_ = position.relativePos;
-    this.updateRelativePos_(callback);
+    this.observer_.updateRelativePos(this);
   }
 
   /**
    * Called when a position change is detected on the injected
    * bottom sentinel element
    * @param {?PositionInViewportEntryDef} position
-   * @param {function(!ViewportRelativePos)} callback
    */
-  bottomSentinelPositionChanged(position, callback) {
-    const prevBottomSentinelPosition = this.bottomSentinelPosition_;
-    if (!position || position.relativePos === prevBottomSentinelPosition) {
-      return;
-    }
+  bottomSentinelPositionChanged(position) {
```

Comment:
Very long public method name, why is it public? :)
Isn't the concept of a `sentinel` very local?

Reply:
Changed